File: Study Plan/Algorithm I/First_Bad_Version.cpp
// The API isBadVersion is defined for you.
// bool isBadVersion(int version);

class Solution {
public:
    int firstBadVersion(int n) {
        int low = 1, high = n;
        int mid, result = 1;
        while(low <= high)
        {
            mid = low + (high - low) / 2;
            if(isBadVersion(mid))
            {
                result = mid;
                high = mid - 1;
            }
            else
                low = mid + 1;   
        }
        return result;
    }
};

File: Study Plan/Algorithm I/First_Bad_Version.py
# The isBadVersion API is already defined for you.
# @param version, an integer
# @return an integer
# def isBadVersion(version):

class Solution:
    def firstBadVersion(self, n):
        """
        :type n: int
        :rtype: int
        """
        low, high = 1, n
        result = 1
        while low <= high:
            mid = low + (high - low) // 2
            if isBadVersion(mid):
                result = mid
                high = mid - 1;
            else:
                low = mid + 1
        return result
            
        